### ISSUE TITLE
Allow ne to be used to compare null against a string

### ISSUE THREAD

**Issue by charles-dyfis-net** [hairyhenderson/gomplate#693]
Right now, comparing a null value against a string results in a fatal error. Consider the following command-line usage:

```
$ gomplate -d userInput=stdin:///in.json --file repro.tmpl <<<'{"key": null}'
template: repro.tmpl:3:6: executing "repro.tmpl" at <ne $dictWithDefaults.key "">: error calling ne: invalid type for comparison
```

...with the template:

```
{{- $userInput := ds "userInput" -}}
{{- $dictWithDefaults := merge $userInput (dict "key" "") -}}
{{ if ne $dictWithDefaults.key "" }}
The user-provided input is: {{$userInput.key}}
{{ else }}
No user-provided input was given.
{{ end }}
```

I would expect `ne` to return true when (exactly/only) one argument is null, rather than to fail to evaluate. (Thus, I would expect output of: `The user-provided input is: <no value>`)

**Comment by hairyhenderson** [comment]
Hi @charles-dyfis-net!

`ne` is a built-in function (see [the Go text/template docs](https://golang.org/pkg/text/template/#hdr-Functions)), and unfortunately they've chosen these particular semantics:

> The comparison functions work on basic types only (or named basic types, such as "type Celsius float32"). They implement the Go rules for comparison of values, except that size and exact type are ignored, so any integer value, signed or unsigned, may be compared with any other integer value. (The arithmetic value is compared, not the bit pattern, so all negative integers are less than all unsigned integers.) However, as usual, one may not compare an int with a float32 and so on.

tl;dr: it won't allow you to compare a `nil` with a `string`.

Another approach could be to use [`conv.ToString`](https://docs.gomplate.ca/functions/conv/#conv-tostring) first to coerce the `nil` into a `string` before the comparison:

```
{{ if ne (conv.ToString $dictWithDefaults.key) "" }}
```

Though now that I think of it, if you're always wanting to compare it to an _empty_ string, you don't even need `eq`, since `if` considers any "empty" value to be `false`:

> The empty values are false, 0, any nil pointer or interface value, and any array, slice, map, or string of length zero.

So, this would work in your example:

```
{{ if $dictWithDefaults.key }}
No user-provided input was given.
{{ else }}
The user-provided input is: {{$userInput.key}}
{{ end }}
```

Or you could keep things in the same order with the `not` function:

```
{{ if not $dictWithDefaults.key }}
The user-provided input is: {{$userInput.key}}
{{ else }}
No user-provided input was given.
{{ end }}
```

Hopefully this helps!

**Comment by hairyhenderson** [comment]
@charles-dyfis-net since I haven't heard anything for almost a month I'm going to assume that this problem is solved and my suggestions worked for you - feel free to re-open if not! 🙂
